src/python_be.cc

```diff
@@ -1361,18 +1361,24 @@ ModelInstanceState::ProcessRequestsDecoupled(
   reporter.SetBatchStatistics(request_count);
 
   if (response_batch.data_->has_error) {
+    for (auto& infer_request : pb_infer_requests) {
+      // Reset the release flags for all the requests.
+      infer_request->SetReleaseFlags(TRITONSERVER_REQUEST_RELEASE_ALL);
+      // Clean up the response factory map.
+      {
+        std::lock_guard<std::mutex> guard{response_factory_map_mutex_};
+        response_factory_map_.erase(
+            reinterpret_cast<intptr_t>(infer_request->RequestAddress()));
+      }
```

Comment:
Would it make sense to encapsulate this in some `infer_request->Reset()` ?

Reply:
I think having them as is makes more sense to me since `response_factory_map_` and the mutex are not part of `InferRequest`, but open to any suggestion.

Reply:
Ah didn't look too closely, makes sense as is.

Reply:
Makes sense! Encapsulated this in a separate function. I think it makes more sense to move that function to the outer scope so that the requests and response factory can be cleaned up at one place.

> Are the request objects re-used or something here? Curious why we are "resetting" release flags in general.

The request objects will be used at the end of `TRITONBACKEND_ModelInstanceExecute` to get the flags when calling `TRITONBACKEND_RequestRelease`. Resetting the flags here because the flags may be set to "rescheduled" in the model.